### PR TITLE
Fixed data_load

### PR DESCRIPTION
**Pull request by lehatrutenb**

### CODE REVIEW

File: AutoML2.0.py [hunk 1]
```diff
@@ -31,118 +31,106 @@
...
 
-all_res = {}
+TRAIN = 6171
+EXAM = 500
 
 def get_data():
-    df = pd.DataFrame({'Quantity': [], 'Best_Minrun': [], 'Best_Time': []})
+    
+    def get_arr_by_str(s):
+        return s.replace('[', '').replace(']', '').replace(',', '').split()
```
Comment (sooobus) [inline AutoML2.0.py:42]: def get_arr_convert_to_int(s):
   s = get_arr_by_str(s)
   return list(map(int, s))

File: AutoML2.0.py [hunk 2]
```diff
@@ -31,118 +31,106 @@
...
+
+    def get_data_from_file(filename, path):
+        db = pd.read_csv(path + '/' + file)
+        df = pd.DataFrame({'Quantity': [], 'Best_Minrun': [], 'Best_Time': []})
+        for i in range(len(db['Best_minrun'])):
+            df['Best_Minrun'][i] = get_arr_by_str(db['Best_minrun'][i])
+
+        for i in range(len(df['Best_Minrun'])):
+            for j in range(len(df['Best_Minrun'][i])):
+                df['Best_Minrun'][i][j] = int(df['Best_Minrun'][i][j])
```
Comment (sooobus) [inline AutoML2.0.py:52]: Тогда эти два цикла будут объединены

File: AutoML2.0.py [hunk 3]
```diff
@@ -31,118 +31,106 @@
...
+        
+        local_train_pairs = {}
+        for i in range(len(df['Quantity'])):
+            pair = float(df['Quantity'][i]), float(df['Best_Minrun'][i][0])
+            local_train_pairs[pair] = 0
+        return local_train_pairs
+    
     pd.options.mode.chained_assignment = None
+    # path = sys.argv[0]
     path = "/home/leha/Desktop/projects/all_timsort_data"
```
Comment (sooobus) [inline AutoML2.0.py:68]:
```
if len(sys.argv) > 1:
  path = sys.argv[1]
else:
  path = "path/to/file"
```

File: AutoML2.0.py [hunk 4]
```diff
@@ -31,118 +31,106 @@
...
-
-data, data_labels = np.asarray([pair[0] for pair in pairs]), np.asarray([pair[1] for pair in pairs])
-
-train_const = 6171
-exam_len_const = 500
-
-train = data[:train_const]
-train_labels = data_labels[:train_const]
+def get_parsed_data():
+
```
Comment (sooobus) [inline AutoML2.0.py:80]: no whiteline (check with pep8, lint)

File: AutoML2.0.py [hunk 5]
```diff
@@ -31,118 +31,106 @@
...
-train = data[:train_const]
-train_labels = data_labels[:train_const]
+def get_parsed_data():
+    
+    def normalize_data(mean, std, data):
+        data -= mean
+        data /= std
+    
+    def divide_data(data):
+        return [data[:floor(len(data) / 4)],
```
Comment (sooobus) [inline AutoML2.0.py:86]: len(data) // 4

File: AutoML2.0.py [hunk 6]
```diff
@@ -31,118 +31,106 @@
...
+                data[floor(len(data) / 4):floor(len(data) / 2)], 
+                data[floor(len(data) / 2):-floor(len(data) / 4)], 
+                data[-floor(len(data) / 4):]]
+    
+    pairs = get_data()
+    pairs = sorted([*pairs.keys()])
 
-exam = np.asarray([])
-exam_labels = np.asarray([])
+    data, data_labels = np.asarray([pair[0] for pair in pairs]), np.asarray([pair[1] for pair in pairs])
```
Comment (sooobus) [inline AutoML2.0.py:94]: лучше разбить на две строки

File: AutoML2.0.py [hunk 7]
```diff
@@ -31,118 +31,106 @@
...
 
-test_p = np.asarray([])
-test_labels_p = np.asarray([])
+    train, train_labels = data[:TRAIN], data_labels[:TRAIN]
```
Comment (sooobus) [inline AutoML2.0.py:96]: shuffle data, maybe sklearn.train_test_split

File: AutoML2.0.py [hunk 8]
```diff
@@ -31,118 +31,106 @@
...
-    else:
-        if len(exam) < exam_len_const and i % 3 != 0:
-            exam = np.append(exam, data[i])
-            exam_labels = np.append(exam_labels, data_labels[i])
+    exam, exam_labels = np.asarray([]), np.asarray([])
+    test_p, test_labels_p = np.asarray([]), np.asarray([])
+    train_p, train_labels_p = np.asarray([]), np.asarray([])
+    
+    left, right = TRAIN + 1, len(data)
+    for i in range(left, right):
```
Comment (sooobus) [inline AutoML2.0.py:103]: чтобы разделить на две выборки, можно сделать еще раз train_test_split

File: AutoML2.0.py [hunk 9]
```diff
@@ -31,118 +31,106 @@
...
-train -= mean
-std = train.std(axis=0) # std - стандартное отклонение = sqrt(mean(abs(x - x.mean())**2)) mean - среднее арифметическое
-train /= std
-
-test -= mean
-test /= std
-
-exam -= mean
-exam /= std
+            if len(exam) < EXAM and i % 3 != 0:
```
Comment (sooobus) [inline AutoML2.0.py:108]: EVALUATION or EVAL

File: AutoML2.0.py [hunk 10]
```diff
@@ -31,118 +31,106 @@
...
 
-data_p = {'mean': mean, 'std': std}
-with open('data.json', 'w') as f_write:
-    json.dump(data_p, f_write)
+    mean = train.mean(axis=0)
+    train -= mean
+    std = train.std(axis=0)
+    train /= std
+    
+    normalize_data(mean, std, test)
```
Comment (sooobus) [inline AutoML2.0.py:124]: new_test = normalize_data(mean, std, test)
и внутри функции тоже можно изменить

**Review by sooobus — CHANGES_REQUESTED** [review]
